aspnetcore/fundamentals/middleware/rate-limit/samples/WebRateLimitAuth/Program.cs

```diff
@@ -0,0 +1,197 @@
+          (remoteIPaddress, key => new SlidingWindowRateLimiterOptions(
+            permitLimit: 10,
+            queueProcessingOrder: QueueProcessingOrder.OldestFirst,
+            queueLimit: 0,
+            window: TimeSpan.FromSeconds(30),
+            segmentsPerWindow: 1
+            ));
+    }
+    else
+    {
+        return RateLimitPartition.CreateNoLimiter<IPAddress>(IPAddress.Loopback);
+    }
+});
+
+app.UseRateLimiter(options);
+
+app.MapRazorPages().RequireRateLimiting(userPolicyName);
+
+static string GetUserEndPoint(HttpContext context) => 
+    $"Hello {context.User?.Identity?.Name ?? "Anonymous"}  {context.Request.Path}";
+
+app.MapGet("/a", (HttpContext context) => $"{GetUserEndPoint(context)}")
+    .RequireRateLimiting(userPolicyName);
+
+app.MapGet("/b", (HttpContext context) => $"{GetUserEndPoint(context)}")
+    .RequireRateLimiting(userPolicyName);
+
+app.MapGet("/c", (HttpContext context) => $"{GetUserEndPoint(context)}");
+
+app.Run();
+// </snippet>
+// </snippet_1>
+#elif ADMIN
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.RateLimiting;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Primitives;
+using System.Threading.RateLimiting;
+using WebRateLimitAuth.Data;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddAuthorization(o => o.AddPolicy("AdminsOnly",
+                                  b => b.RequireClaim("admin", "true")));
+
+var connectionString = builder.Configuration.GetConnectionString("DefaultConnection") 
+    ?? throw new InvalidOperationException("Connection string 'DefaultConnection' not found.");
+builder.Services.AddDbContext<ApplicationDbContext>(options =>
+    options.UseSqlServer(connectionString));
+builder.Services.AddDatabaseDeveloperPageExceptionFilter();
+
+builder.Services.AddDefaultIdentity<IdentityUser>(options => 
+                  options.SignIn.RequireConfirmedAccount = true)
+    .AddEntityFrameworkStores<ApplicationDbContext>();
+builder.Services.AddRazorPages();
+
+var app = builder.Build();
+
+if (app.Environment.IsDevelopment())
+{
+    app.UseMigrationsEndPoint();
+}
+else
+{
+    app.UseExceptionHandler("/Error");
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+app.UseStaticFiles();
+
+app.UseRouting();
+
+// <snippet_adm>
+app.UseAuthentication();
+app.UseAuthorization();
+
+var getPolicyName = "get";
+var adminPolicyName = "admin";
+var postPolicyName = "post";
+
+app.UseRateLimiter(new RateLimiterOptions()
+    .AddConcurrencyLimiter(policyName: getPolicyName,
+          new ConcurrencyLimiterOptions(permitLimit: 2,
+          queueProcessingOrder: QueueProcessingOrder.OldestFirst, queueLimit: 2))
+    .AddNoLimiter(policyName: adminPolicyName)
+    .AddPolicy(policyName: postPolicyName, partitioner: httpContext =>
+    {
+        if (!StringValues.IsNullOrEmpty(httpContext.Request.Headers["my-token"]))
+        {
+            return RateLimitPartition.CreateTokenBucketLimiter("token", key =>
```

Comment:
Is this suppose to limit based on each individual token or just one limiter for all tokens. If it's the first one I think the code needs to change.
```suggestion
            return RateLimitPartition.CreateTokenBucketLimiter(httpContext.Request.Headers["my-token"], key =>
```